File: gestor-reservas-academicas-back/src/main/java/com/servigestor360/service/ReservaService.java
package com.servigestor360.service;

import com.servigestor360.entity.Reserva;
import com.servigestor360.repository.ReservaRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * Servicio encargado de contener la lógica de negocio
 * relacionada con las reservas académicas.
 */
@Service
public class ReservaService {

    @Autowired
    private ReservaRepository reservaRepository;

    /**
     * Obtiene todas las reservas registradas.
     */
    public List<Reserva> listarReservas() {
        return reservaRepository.findAll();
    }

    /**
     * Guarda una nueva reserva.
     */
    public Reserva guardarReserva(Reserva reserva) {
        return reservaRepository.save(reserva);
    }

    /**
     * Busca una reserva por su identificador.
     */
    public Reserva buscarPorId(Integer id) {
        return reservaRepository.findById(id).orElse(null);
    }

    /**
     * Actualiza la información de una reserva existente.
     */
    public Reserva actualizarReserva(Integer id, Reserva reservaActualizada) {

        Reserva reserva = reservaRepository.findById(id).orElse(null);

        if (reserva != null) {

            reserva.setNombreSolicitante(reservaActualizada.getNombreSolicitante());
            reserva.setFechaReserva(reservaActualizada.getFechaReserva());
            reserva.setHoraInicio(reservaActualizada.getHoraInicio());
            reserva.setHoraFin(reservaActualizada.getHoraFin());
            reserva.setEstado(reservaActualizada.getEstado());
            reserva.setSala(reservaActualizada.getSala());

            return reservaRepository.save(reserva);
        }

        return null;
    }

    /**
     * Elimina una reserva por su identificador.
     */
    public void eliminarReserva(Integer id) {
        reservaRepository.deleteById(id);
    }
}

File: gestor-reservas-academicas-front/src/services/reservaService.js
// URL base del controlador de Reservas en el backend
const URL = "http://localhost:8080/reservas";

// Obtiene todas las reservas registradas
export async function obtenerReservas() {

    const respuesta = await fetch(URL);

    return await respuesta.json();

}

// Envía una nueva reserva al servidor
export async function guardarReserva(reserva) {

    const respuesta = await fetch(URL, {

        method: "POST",

        headers: {
            "Content-Type": "application/json"
        },

        body: JSON.stringify(reserva)

    });

    return await respuesta.json();

}

// Actualiza la información de una reserva existente
export async function actualizarReserva(id, reserva) {

    const respuesta = await fetch(`${URL}/${id}`, {

        method: "PUT",

        headers: {
            "Content-Type": "application/json"
        },

        body: JSON.stringify(reserva)

    });

    return await respuesta.json();

}

// Elimina una reserva según su identificador
export async function eliminarReserva(id) {

    await fetch(`${URL}/${id}`, {

        method: "DELETE"

    });

}